test/algorithms/closest_points/empty_geometry.hpp

```diff
@@ -0,0 +1,267 @@
+// Boost.Geometry
+// Unit Test
+
+// Copyright (c) 2021, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_TEST_ALGORITHMS_CLOSEST_POINTS_EMPTY_GEOMETRY_HPP
+#define BOOST_GEOMETRY_TEST_ALGORITHMS_CLOSEST_POINTS_EMPTY_GEOMETRY_HPP
+
+template <typename Point, typename Geometry1, typename Geometry2,
+          typename Strategies>
+void test_empty_input(Geometry1 const& geometry1,
+                      Geometry2 const& geometry2,
+                      Strategies const& strategies)
+{
+    bg::model::segment<Point> out_seg;
+
+    try
+    {
+        bg::closest_points(geometry1, geometry2, out_seg);
+    }
+    catch(bg::empty_input_exception const& )
+    {
+        return;
+    }
+    BOOST_CHECK_MESSAGE(false,
+                        "A empty_input_exception should have been thrown");
```

Comment:
Shouldn't exceptions be checked with `BOOST_CHECK_EXCEPTION`?

Reply:
probably yes, but in the whole library they are checked like this so I just follow that.